src/omv/modules/py_tf.c

```diff
@@ -803,31 +720,382 @@ STATIC void py_tf_regression_input_callback(void *callback_data,
+    o->output_width = mp_obj_new_int(params->output_width);
+    o->output_channels = mp_obj_new_int(params->output_channels);
+    list_init(&o->bounding_boxes, sizeof(py_tf_model_output_bounding_box_lnk_data_t));
+    *(arg->out) = mp_call_function_1(arg->callback, o);
+}
+
+STATIC mp_obj_t py_tf_predict(uint n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    enum { ARG_roi, ARG_callback, ARG_input_scale, ARG_input_normalization };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_roi, MP_ARG_OBJ | MP_ARG_KW_ONLY, {.u_rom_obj = MP_ROM_NONE} },
+        { MP_QSTR_callback, MP_ARG_OBJ | MP_ARG_KW_ONLY, {.u_rom_obj = MP_ROM_NONE} },
+        { MP_QSTR_input_scale, MP_ARG_INT | MP_ARG_KW_ONLY, {.u_int = PY_TF_SCALE_0_1} },
+        { MP_QSTR_input_normalization, MP_ARG_OBJ | MP_ARG_KW_ONLY, {.u_rom_obj = MP_ROM_NONE} },
+    };
+
+    // Parse args.
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args - 2, pos_args + 2, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
     fb_alloc_mark();
     py_tf_alloc_putchar_buffer();
 
-    py_tf_model_obj_t *model = py_tf_load_alloc(model_obj);
+    py_tf_model_obj_t *model = py_tf_load_alloc(pos_args[0]);
     uint8_t *tensor_arena = fb_alloc(model->params.tensor_arena_size, FB_ALLOC_PREFER_SPEED | FB_ALLOC_CACHE_ALIGN);
 
-    mp_obj_t py_tf_classify_output_callback_data;
+    mp_obj_t output_callback_data;
+    int invoke_result;
 
-    if (libtf_invoke(model->model_data,
-                     tensor_arena,
-                     &model->params,
-                     py_tf_regression_input_callback,
-                     &array_obj,
-                     py_tf_classify_output_callback,
-                     &py_tf_classify_output_callback_data) != 0) {
+    if (MP_OBJ_IS_TYPE(pos_args[1], &mp_type_tuple) || MP_OBJ_IS_TYPE(pos_args[1], &mp_type_list)) {
+        invoke_result = libtf_invoke(model->model_data,
+                                     tensor_arena,
+                                     &model->params,
+                                     py_tf_regression_input_callback,
+                                     (void *) &pos_args[1],
+                                     py_tf_output_callback,
+                                     &output_callback_data);
+    } else {
+        image_t *image = py_helper_arg_to_image(pos_args[1], ARG_IMAGE_ANY);
+        rectangle_t roi = py_helper_arg_to_roi(args[ARG_roi].u_obj, image);
+
+        py_tf_input_callback_data_t py_tf_input_callback_data =
+        {.input_normalization = {0.0f, 0.0f, 0.0f, 1.0f, 1.0f, 1.0f}};
+        py_tf_input_callback_data.img = image;
+        py_tf_input_callback_data.roi = &roi;
+        py_tf_input_callback_data.input_scale = args[ARG_input_scale].u_int;
+        py_helper_arg_to_float_array(args[ARG_input_normalization].u_obj,
+                                     py_tf_input_callback_data.input_normalization, PY_TF_INPUT_NORMALIZATION_SIZE);
```

Comment:
Should be the same as `py_tf_segement` please see review comments above.

Reply:
This is an outdated comment. I changed it previously to what was discussed in the meeting with mean/std.